requirements.txt

```diff
@@ -1,11 +1,14 @@
+cycler==0.10.0
 h5py==2.10.0
-numpy==1.17.3
+kiwisolver==1.2.0
```

Comment:
Looks like in addition to resolving conflicting versions, some packages were moved from notebook_requirements.txt to requirements.txt (kiwisolver, cycler, patsy).  `pip show` tells me they are all required by matplotlib, which isn't just used in the notebook, so it makes sense to include them in the main requirements file.  Is that the reasoning?

Reply:
Yep, that is the reasoning. Following https://blog.miguelgrinberg.com/post/the-package-dependency-blues, let's keep indirect requirements in `requirements.txt`